go/private/context.bzl

```diff
@@ -291,7 +296,6 @@ def _library_to_source(go, attr, library, coverage_instrumented):
         "cgo_deps": [],
         "cgo_exports": [],
         "cc_info": None,
-        "pgoprofile": getattr(attr, "pgoprofile", None),
```

Comment:
I think `pgoprofile` was never used

Reply:
That would be concerning as it's a pretty new feature. Are you sure? Unfortunately we didn't find a way to cover it in tests, so we can't rely on them to catch any regressions.

Reply:
I originally missed that this attribute could be specified on `go_binary`, so I thought this was always `None`. I am still not seeing where we read this attribute off the provider (I think it gets set on the cmd line through a transition), but I'll remove this from the PR to be safe